src/app/index.js

```diff
@@ -109,19 +114,70 @@ export default class extends Generator {
 
     this.registerTransformStream(
       rename(path => {
+        const extension = this.props.useTypeScript ? '.ts' : '.js'
         if (path.basename === 'index' && !path.extname) {
-          path.extname = '.js'
+          path.extname = extension
         }
 
         if (path.extname === '.test') {
           path.basename += '.test'
-          path.extname = '.js'
+          path.extname = extension
         }
 
         return path
       })
     )
 
+    const platformProperties = {}
+    if (this.props.useTypeScript) {
+      platformProperties[
+        'scripts'
+      ] = Object.assign(defaultPackageJson.scripts, {
+        build: 'tsc',
+        prettier: 'prettier',
+        'prettier-write': 'npm run prettier -- --parser typescript --no-semi --trailing-comma es5 --write --print-width 120',
+        'prettier-project': "npm run prettier-write -- 'src/**/*.{ts,tsx}'",
```

Comment:
Bug did get fixed https://github.com/yarnpkg/yarn/pull/3627, so in a later version this can be updated